util/app-config/src/cli.rs

```diff
@@ -526,3 +527,18 @@ fn is_hex(hex: &str) -> Result<(), String> {
         Err("Must 0x-prefixed hexadecimal string".to_string())
     }
 }
+
+fn is_h256(hex: &str) -> Result<(), String> {
+    if hex.as_bytes()[..2] == b"0x"[..] {
+        let mut hd = [0u8; 32];
+        if let Err(e) = faster_hex::hex_decode(&hex.as_bytes()[2..], &mut hd) {
+            return Err(e.to_string());
+        };
+        match H256::from_slice(&hd) {
+            Ok(_) => Ok(()),
+            _ => Err("input string is not valid H256 format".to_owned()),
+        }
+    } else {
+        Err("Must be a 0x-prefixed hexadecimal string".to_owned())
+    }
```

Comment:
```suggestion
    is_hex(hex) && hex.len == 34
```

Reply:
it seems it' not so good to check byte array length match 34 to decide h256 string.
here is a little issue:  in mainnet, height [6749464](https://explorer.nervos.org/block/6749464), block hash is `0x00d1b86f6824d33a91b72ec20e2118cf7788a5ffff656bd1ea1ea638c764cb5f`
we use `is_hex(hex) && hex.len == 34`, it works and works good.

but if we input  `0xd1b86f6824d33a91b72ec20e2118cf7788a5ffff656bd1ea1ea638c764cb5f`, the hex.len == 34 will fail, but it's hard to say the `0xd1b86f6824d33a91b72ec20e2118cf7788a5ffff656bd1ea1ea638c764cb5f` is incorrect.
actually, i did think about it for a while, and not sure about it.

Reply:
let's keep it simple stupid, user won’t copy a hex string and delete leading or trailing zeros manually

Reply:
ok. decided for now.